contracts/test/StakeTable.t.sol

```diff
@@ -89,12 +104,20 @@ contract StakeTable_Test is Test {
             sig,
             validUntilEpoch
         );
-
-        assertTrue(res);
+        if (!expectRevert) {
+            assertTrue(res);
+        } else {
+            assertFalse(res);
+        }
 
         return (blsVK, depositAmount);
     }
 
+    /// @dev  Helper function to simulate a successful registration
+    function runSuccessfulRegistration() private returns (BN254.G2Point memory, uint256) {
+        return registerWithSeed(exampleTokenCreator, 34, 10, false);
+    }
```

Comment:
I saw almost all subsequent functions starts with this, why don't we add this in `setUp()` so that we don't have to manually call this in each test case?

Reply:
It makes each test harder to understand I think, especially because some tests would not need to use `runSuccessfulRegistration` (e.g. `test_RevertWhen_TransferFailed`).

Reply:
why? just having some prior successful registration is perfectly fine? it's equivalent of initializing a ledger with a non-empty stake table to start with (before the logic of each tests start) 

`test_RevertWhen_TransferFailed` shouldn't be affected, right? it'll still pass even with the addition in `setUp()`, correct?

Reply:
The test test_`RevertWhen_TransferFailed` fails on this condition:

```
assertEq(ERC20(token).balanceOf(exampleTokenCreator), INITIAL_BALANCE);
```

It is easy to fix (by replacing INITIAL_BALANCE with `INITIAL_BALANCE-10` but it would be harder to read/understand I think.  

> it's equivalent of initializing a ledger with a non-empty stake table to start with (before the logic of each tests start)

What I don't like too much about this is that the test needs to know about the setup. For example you cannot reuse the same seed to generate the wallet in `test_Registration_succeeds`, otherwise the registration will fail.
It feels simpler to start with a blank state and prepare this state inside the test function. I would prioritize being explicit in the test function over minimizing code duplication in this case.

Reply:
okay, sure. 

>What I don't like too much about this is that the test needs to know about the setup.

altho for this point, you are supposed to know what happens to `setUp`, the first thing you read each test is to read setUp. so I don't think that's a problem.
That said, it's not too much code duplication. i don't have strong opinion here.